### PR TITLE
Move Cleanup.h from include/tscpp/api to include/ts.

### PR DESCRIPTION
**Pull request by ywkaras**
Change the namespace of its declarations from atscppapi to tsapi::c_support. This is in support of removing the main, high-overhead parts of the ATS C++ API.

I used `git mv` to move the file, frustrating that github shows it as a deletion and a new file.

### CODE REVIEW

**Comment by maskit** [comment]
Does this mean everything in Cleanup.h is going to be part of TS API and require API proposal and deprecation period when we change it? Or is it still just an utility that does not guarantee any compatibility between ATS versions?

**Comment by ywkaras** [comment]
> Does this mean everything in Cleanup.h is going to be part of TS API and require API proposal and deprecation period when we change it? Or is it still just an utility that does not guarantee any compatibility between ATS versions?

Yes I think so.

**Comment by cmcfarlen** [comment]
Should this go in the new `include/api` directory?

**Comment by ywkaras** [comment]
> Should this go in the new `include/api` directory?

I haven't heard about that directory, maybe I wasn't in the right meeting (or I slipped into a meeting coma).

**Comment by maskit** [comment]
> > Does this mean everything in Cleanup.h is going to be part of TS API and require API proposal and deprecation period when we change it? Or is it still just a utility that does not guarantee any compatibility between ATS versions?

> Yes I think so.

Which one?

**Comment by maskit** [comment]
> It will be part of the TS API.

Then it's an addition in terms of TS API, and a proposal need to be sent on the dev list.

**Comment by ywkaras** [comment]
I have brought it up on the mailing list.

**Comment by maskit** [comment]
You mean the one you sent last year?

File: include/ts/Cleanup.h [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+#pragma once
+
+#include <type_traits>
+#include <memory>
+
+#include <ts/ts.h>
+
+namespace tsapi
+{
+// This sub-namespace is for declarations specifically designed to work with declarations in tsapi::c.
```
Comment (maskit) [inline include/ts/Cleanup.h:33]: Where's `tsapi::c`?

Comment (ywkaras) [reply include/ts/Cleanup.h:33]: In ts.h.

**Comment by ywkaras** [comment]
> You mean the one you sent last year?

https://lists.apache.org/list?dev@trafficserver.apache.org:2023-8

File: include/ts/Cleanup.h (hunk 1 above)
Comment (maskit) [reply include/ts/Cleanup.h:33]: Ok, so it was added recently. I didn't know such namespace was added. I don't think I saw the proposal on the list.

**Comment by ywkaras** [comment]
> You proposed removing C++ API and saving Cleanup.h, but promoting Cleanup.h to TS API is not part of the proposal. You might have meant it, but I didn't think that is included. And the thread is about "Removing example/plugins/cpp-api" and it's not even started by you. I don't think it's a valid proposal.

https://lists.apache.org/list?dev@trafficserver.apache.org:2023-8

**Pull request closed without merging**